<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Changelog • regionReport</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Changelog" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-news">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">regionReport</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.27.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/regionReport.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/bumphunterExample.html">Example report using bumphunter results</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/leekgroup/regionReport/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href='https://github.com/leekgroup/regionReport/blob/master/NEWS.md'><code>NEWS.md</code></a></small>
    </div>

    <div id="regionreport-1232" class="section level1">
<h1 class="page-header" data-toc-text="1.23.2">
<a href="#regionreport-1232" class="anchor"></a>regionReport 1.23.2</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>No longer suggest <code>DESeq</code> since this package has been deprecated.</li>
</ul>
</div>
    <div id="regionreport-12111" class="section level1">
<h1 class="page-header" data-toc-text="1.21.11">
<a href="#regionreport-12111" class="anchor"></a>regionReport 1.21.11</h1>
<p>BUG FIXES</p>
<ul>
<li>Cropping images through <code>magick::image_trim()</code> as done by default by <code><a href="https://rdrr.io/pkg/BiocStyle/man/html_document.html">BiocStyle::html_document()</a></code> can fail on Linux. This could be an ImageMagick issue or an issue about lack of resources. The full investigative report is at <a href="https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016650.html" class="uri">https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016650.html</a>. This is related to: <a href="https://github.com/yihui/knitr/issues/1785#issuecomment-574723631" class="uri">https://github.com/yihui/knitr/issues/1785#issuecomment-574723631</a> <a href="https://github.com/yihui/knitr/issues/1796" class="uri">https://github.com/yihui/knitr/issues/1796</a> <a href="https://github.com/Bioconductor/BiocStyle/issues/65#issuecomment-552832630" class="uri">https://github.com/Bioconductor/BiocStyle/issues/65#issuecomment-552832630</a> <a href="https://github.com/ropensci/magick/issues/171" class="uri">https://github.com/ropensci/magick/issues/171</a> <a href="https://github.com/ropensci/magick/issues/194" class="uri">https://github.com/ropensci/magick/issues/194</a> In regionReport version 1.21.10 I have opted by using <code>crop = NULL</code> to disable cropping of images by <code><a href="https://rdrr.io/pkg/BiocStyle/man/html_document.html">BiocStyle::html_document()</a></code> and thus avoid the issues with <code>ImageMagick</code> either coming from <code>magick</code>, from the version of <code>ImageMagick</code> installed on the Linux Bioconductor build machine and devel docker, or from resources in these two Linux environments as described in the investigative report. About a month ago I also saw failures on Windows on Bioc 3.10. Whether they were caused by ggbio 1.35.1 or this issue will remain a mystery. But it’s likely that this <code>magick::image_trim()</code> issue also affected the Bioconductor windows builder. The related bioc-devel threads are: <a href="https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016538.html" class="uri">https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016538.html</a> <a href="https://stat.ethz.ch/pipermail/bioc-devel/2020-March/016365.html" class="uri">https://stat.ethz.ch/pipermail/bioc-devel/2020-March/016365.html</a>
</li>
</ul>
</div>
    <div id="regionreport-12110" class="section level1">
<h1 class="page-header" data-toc-text="1.21.10">
<a href="#regionreport-12110" class="anchor"></a>regionReport 1.21.10</h1>
<p>BUG FIXES</p>
<ul>
<li>Add <code>crop = NULL</code> to all template Rmarkdown files. Drop <code>magick</code> as a depencency since cropping is not used on the reports (as done through <code>BiocStyle</code> by default). This is again related to responding to <a href="https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016645.html" class="uri">https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016645.html</a>.</li>
</ul>
</div>
    <div id="regionreport-1219" class="section level1">
<h1 class="page-header" data-toc-text="1.21.9">
<a href="#regionreport-1219" class="anchor"></a>regionReport 1.21.9</h1>
<p>BUG FIXES</p>
<ul>
<li>Ran another test in preparation for <a href="https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016645.html" class="uri">https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016645.html</a> after the test in version 1.21.8 failed. The code will now be run in the example of <code><a href="../reference/derfinderReport.html">derfinderReport()</a></code> instead of the vignette.</li>
</ul>
</div>
    <div id="regionreport-1218" class="section level1">
<h1 class="page-header" data-toc-text="1.21.8">
<a href="#regionreport-1218" class="anchor"></a>regionReport 1.21.8</h1>
<p>BUG FIXES</p>
<ul>
<li>In an attempt to fix the bug I reported at <a href="https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016645.html" class="uri">https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016645.html</a> and in preparation to reply to this thread, I’m experimenting with suggesting that <code>magick</code> is installed, at which point <code>BiocStyle</code> will try to crop. Another option might be to use <code>crop = NULL</code> on all Rmd files.</li>
</ul>
</div>
    <div id="regionreport-1215" class="section level1">
<h1 class="page-header" data-toc-text="1.21.5">
<a href="#regionreport-1215" class="anchor"></a>regionReport 1.21.5</h1>
<p>BUG FIXES</p>
<ul>
<li>The example in <code><a href="../reference/renderReport.html">renderReport()</a></code> was failing because the first time the function was called, it ran <code><a href="https://rdrr.io/pkg/derfinder/man/makeGenomicState.html">derfinder::makeGenomicState()</a></code> which in turn uses <code><a href="https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html">GenomicFeatures::isActiveSeq()</a></code>. <code>derfinder</code> version 1.21.5 fixed this bug and is thus required by <code>regionReport</code> now.</li>
</ul>
</div>
    <div id="regionreport-1214" class="section level1">
<h1 class="page-header" data-toc-text="1.21.4">
<a href="#regionreport-1214" class="anchor"></a>regionReport 1.21.4</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Add links to the example reports created when deploying the documentation website with <code><a href="https://pkgdown.r-lib.org/reference/deploy_to_branch.html">pkgdown::deploy_to_branch()</a></code>. This eliminates the need for the <code>vignetttes/Makefile</code> and the fake vignettes. It should also avoid confusing <code>pkgdown</code>.</li>
</ul>
</div>
    <div id="regionreport-1213" class="section level1">
<h1 class="page-header" data-toc-text="1.21.3">
<a href="#regionreport-1213" class="anchor"></a>regionReport 1.21.3</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Documentation website is now available at <a href="http://leekgroup.github.io/derfinderPlot/" class="uri">http://leekgroup.github.io/derfinderPlot/</a>. It gets updated with every commit on the master branch (bioc-devel) using GitHub Actions and pkgdown.</li>
</ul>
</div>
    <div id="regionreport-1212" class="section level1">
<h1 class="page-header" data-toc-text="1.21.2">
<a href="#regionreport-1212" class="anchor"></a>regionReport 1.21.2</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Now use <code>GenomeIndoDb::getChromInfoFromUCSC()</code> instead of <code><a href="https://rdrr.io/r/utils/data.html">data(hg19Ideogram, package = 'biovizBase')</a></code> to get the hg19 chromosome lengths.</li>
</ul>
</div>
    <div id="regionreport-1192" class="section level1">
<h1 class="page-header" data-toc-text="1.19.2">
<a href="#regionreport-1192" class="anchor"></a>regionReport 1.19.2</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Added a <code>NEWS.md</code> file to track changes to the package.</li>
</ul>
</div>
    <div id="regionreport-1192-1" class="section level1">
<h1 class="page-header" data-toc-text="1.19.2">
<a href="#regionreport-1192-1" class="anchor"></a>regionReport 1.19.2</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Renamed <code>load_install()</code> as <code>load_check()</code> as this function now only checks that the package(s) was installed and returns an error if missing. The error shows the user how to install the package(s) they are missing instead of installing them automatically. This complies with Marcel Ramos’ request at <a href="https://github.com/leekgroup/recount/issues/14" class="uri">https://github.com/leekgroup/recount/issues/14</a>.</li>
</ul>
</div>
    <div id="regionreport-1172" class="section level1">
<h1 class="page-header" data-toc-text="1.17.2">
<a href="#regionreport-1172" class="anchor"></a>regionReport 1.17.2</h1>
<p>NEW FEATURES</p>
<ul>
<li>Add ORCID’s following changes at <a href="http://bioconductor.org/developers/package-guidelines/#description" class="uri">http://bioconductor.org/developers/package-guidelines/#description</a>
</li>
</ul>
</div>
    <div id="regionreport-1154" class="section level1">
<h1 class="page-header" data-toc-text="1.15.4">
<a href="#regionreport-1154" class="anchor"></a>regionReport 1.15.4</h1>
<p>BUG FIXES</p>
<ul>
<li>Fix a bug in the order that was reported and fixed by <a href='https://github.com/bounlu'>@bounlu</a> at <a href="https://github.com/leekgroup/regionReport/pull/9/" class="uri">https://github.com/leekgroup/regionReport/pull/9/</a>
</li>
</ul>
</div>
    <div id="regionreport-1153" class="section level1">
<h1 class="page-header" data-toc-text="1.15.3">
<a href="#regionreport-1153" class="anchor"></a>regionReport 1.15.3</h1>
<p>BUG FIXES</p>
<ul>
<li><p>Fixed an issue with <code>DESeq2Exploration.Rmd</code> that affected both <code><a href="../reference/DESeq2Report.html">DESeq2Report()</a></code> and <code><a href="../reference/edgeReport.html">edgeReport()</a></code>. This should also fix the recount bioc-release (3.7) and bioc-devel (3.8) branches.</p></li>
<li><p>Fixed a <code>NAMESPACE</code> issue with <code><a href="https://rdrr.io/pkg/rmarkdown/man/html_document.html">rmarkdown::html_document</a></code> and <code><a href="https://rdrr.io/pkg/BiocStyle/man/html_document.html">BiocStyle::html_document</a></code></p></li>
</ul>
</div>
    <div id="regionreport-1152" class="section level1">
<h1 class="page-header" data-toc-text="1.15.2">
<a href="#regionreport-1152" class="anchor"></a>regionReport 1.15.2</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Use BiocManager</li>
</ul>
</div>
    <div id="regionreport-1151" class="section level1">
<h1 class="page-header" data-toc-text="1.15.1">
<a href="#regionreport-1151" class="anchor"></a>regionReport 1.15.1</h1>
<p>BUG FIXES</p>
<ul>
<li>Fix namespace issue in relation to <code>BiocStyle::html_document2</code>
</li>
</ul>
</div>
    <div id="regionreport-1116" class="section level1">
<h1 class="page-header" data-toc-text="1.11.6">
<a href="#regionreport-1116" class="anchor"></a>regionReport 1.11.6</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Changed the default style to <code><a href="https://rdrr.io/pkg/BiocStyle/man/html_document.html">BiocStyle::html_document</a></code> to mirror recent changes in <code>BiocStyle</code>.</li>
</ul>
</div>
    <div id="regionreport-1114" class="section level1">
<h1 class="page-header" data-toc-text="1.11.4">
<a href="#regionreport-1114" class="anchor"></a>regionReport 1.11.4</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Vignette now uses the new <code><a href="https://rdrr.io/pkg/BiocStyle/man/html_document.html">BiocStyle::html_document</a></code> that was recently released.</li>
</ul>
</div>
    <div id="regionreport-1112" class="section level1">
<h1 class="page-header" data-toc-text="1.11.2">
<a href="#regionreport-1112" class="anchor"></a>regionReport 1.11.2</h1>
<p>BUG FIXES</p>
<ul>
<li>Fixed the citations.</li>
<li>Fixed <code><a href="../reference/DESeq2Report.html">DESeq2Report()</a></code> for <code>limma</code>-results so that it will properly cite <code>limma</code>.</li>
</ul>
</div>
    <div id="regionreport-1111" class="section level1">
<h1 class="page-header" data-toc-text="1.11.1">
<a href="#regionreport-1111" class="anchor"></a>regionReport 1.11.1</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>
<code><a href="../reference/DESeq2Report.html">DESeq2Report()</a></code> can now be used with other software if their results are made to look like <code>DESeq2</code> results. For example, with <code>limma</code>-voom results.</li>
</ul>
<p>BUG FIXES</p>
<ul>
<li>Made the <code><a href="../reference/DESeq2Report.html">DESeq2Report()</a></code> more robust in case<code><a href="https://rdrr.io/pkg/DESeq2/man/rlog.html">rlog()</a></code> fails initially.</li>
</ul>
</div>
    <div id="regionreport-191" class="section level1">
<h1 class="page-header" data-toc-text="1.9.1">
<a href="#regionreport-191" class="anchor"></a>regionReport 1.9.1</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Changed the default style to <code>BiocStyle::html_document2</code>.</li>
</ul>
</div>
    <div id="regionreport-1710" class="section level1">
<h1 class="page-header" data-toc-text="1.7.10">
<a href="#regionreport-1710" class="anchor"></a>regionReport 1.7.10</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Help pages now document advanced arguments.</li>
</ul>
</div>
    <div id="regionreport-172" class="section level1">
<h1 class="page-header" data-toc-text="1.7.2">
<a href="#regionreport-172" class="anchor"></a>regionReport 1.7.2</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Dropped defunct functions.</li>
</ul>
</div>
    <div id="regionreport-1548" class="section level1">
<h1 class="page-header" data-toc-text="1.5.48">
<a href="#regionreport-1548" class="anchor"></a>regionReport 1.5.48</h1>
<p>BUG FIXES</p>
<ul>
<li>Fixed a bug in <code><a href="../reference/derfinderReport.html">derfinderReport()</a></code> for a case when there are significant regions but not all regions have finite areas.</li>
</ul>
</div>
    <div id="regionreport-1543" class="section level1">
<h1 class="page-header" data-toc-text="1.5.43">
<a href="#regionreport-1543" class="anchor"></a>regionReport 1.5.43</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>
<code><a href="../reference/edgeReport.html">edgeReport()</a></code> now includes two edgeR specific plots: one showing the BCV and another showing a 2-dim MDS. Also added more <code>edgeR</code> citations that I missed earlier: thank you Gordon Smyth!</li>
</ul>
</div>
    <div id="regionreport-1533" class="section level1">
<h1 class="page-header" data-toc-text="1.5.33">
<a href="#regionreport-1533" class="anchor"></a>regionReport 1.5.33</h1>
<p>NEW FEATURES</p>
<ul>
<li>Added the function <code><a href="../reference/edgeReport.html">edgeReport()</a></code> for creating HTML or PDF reports based on edgeR results. Together with <code><a href="../reference/DESeq2Report.html">DESeq2Report()</a></code> now <code>regionReport</code> supports the two most used packages for RNA-seq feature-level analysis.</li>
</ul>
</div>
    <div id="regionreport-1519" class="section level1">
<h1 class="page-header" data-toc-text="1.5.19">
<a href="#regionreport-1519" class="anchor"></a>regionReport 1.5.19</h1>
<p>NEW FEATURES</p>
<ul>
<li>Added the templates <code>templatePvalueHistogram</code> and <code>templateHistogram</code> to be used with <code><a href="../reference/renderReport.html">renderReport()</a></code> if you prefer histogram plots instead of density plots.</li>
</ul>
</div>
    <div id="regionreport-1512" class="section level1">
<h1 class="page-header" data-toc-text="1.5.12">
<a href="#regionreport-1512" class="anchor"></a>regionReport 1.5.12</h1>
<p>NEW FEATURES</p>
<ul>
<li>Added the function <code><a href="../reference/DESeq2Report.html">DESeq2Report()</a></code> for creating HTML or PDF reports based on <code>DESeq2</code> results. This should also be useful to explore derfinder results created from the expressed regions-level approach.</li>
</ul>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Added a <code>digits</code> argument to control how to round some numerical variables in all type of reports.</li>
<li>Added a <code>theme</code> argument to allow setting the <code>ggplot2</code> theme for the plots.</li>
</ul>
<p>BUG FIXES</p>
<ul>
<li>Improved the PDF versions of all reports by hiding code and shortening tables. Also added a warning to switch the device to <code>pdf</code> for PDF output since it looks better than the default <code>png</code>.</li>
</ul>
</div>
    <div id="regionreport-156" class="section level1">
<h1 class="page-header" data-toc-text="1.5.6">
<a href="#regionreport-156" class="anchor"></a>regionReport 1.5.6</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Switched to using <code>rmarkdown</code> instead of <code>knitrBootstrap</code> as the default engine for creating the reports.</li>
</ul>
</div>
    <div id="regionreport-138" class="section level1">
<h1 class="page-header" data-toc-text="1.3.8">
<a href="#regionreport-138" class="anchor"></a>regionReport 1.3.8</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>
<code><a href="../reference/renderReport.html">renderReport()</a></code> and <code><a href="../reference/derfinderReport.html">derfinderReport()</a></code> now show Manhattan plots for p-value variables (p-value, q-value, FWER adjusted p-value).</li>
</ul>
</div>
    <div id="regionreport-137" class="section level1">
<h1 class="page-header" data-toc-text="1.3.7">
<a href="#regionreport-137" class="anchor"></a>regionReport 1.3.7</h1>
<p>NEW FEATURES</p>
<ul>
<li>
<code><a href="../reference/renderReport.html">renderReport()</a></code> now has the <code>densityTemplates</code> argument via which users can customize the density plots for the p-value variables and the continuous variables. This addresses one of David Robinson’s requests at <a href="http://f1000research.com/articles/4-105/v1" class="uri">http://f1000research.com/articles/4-105/v1</a>
</li>
</ul>
</div>
    <div id="regionreport-136" class="section level1">
<h1 class="page-header" data-toc-text="1.3.6">
<a href="#regionreport-136" class="anchor"></a>regionReport 1.3.6</h1>
<p>NEW FEATURES</p>
<ul>
<li>Added a vignette with an example report from bumphunter results.</li>
</ul>
</div>
    <div id="regionreport-135" class="section level1">
<h1 class="page-header" data-toc-text="1.3.5">
<a href="#regionreport-135" class="anchor"></a>regionReport 1.3.5</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Merged pull request <a href="https://github.com/leekgroup/regionReport/pull/7" class="uri">https://github.com/leekgroup/regionReport/pull/7</a>
</li>
<li>Added <code>template</code> argument to <code><a href="../reference/renderReport.html">renderReport()</a></code> and <code><a href="../reference/derfinderReport.html">derfinderReport()</a></code> to customize the <code>knitr</code> template used</li>
<li>Wrapped code that works in a temporary directory in <code>with_wd()</code> function, which evaluates in the directory but returns to the original directory in the case of a user interrupt or error (with <code><a href="https://rdrr.io/r/base/on.exit.html">on.exit()</a></code>)</li>
</ul>
</div>
    <div id="regionreport-134" class="section level1">
<h1 class="page-header" data-toc-text="1.3.4">
<a href="#regionreport-134" class="anchor"></a>regionReport 1.3.4</h1>
<p>NEW FEATURES</p>
<ul>
<li>Reports now have a link to the BibTeX file used for the references. This addresses <a href="http://f1000research.com/articles/4-105/v1#reflist" class="uri">http://f1000research.com/articles/4-105/v1#reflist</a> Karthik Ram’s bullet point number 4.</li>
</ul>
</div>
    <div id="regionreport-133" class="section level1">
<h1 class="page-header" data-toc-text="1.3.3">
<a href="#regionreport-133" class="anchor"></a>regionReport 1.3.3</h1>
<p>NEW FEATURES</p>
<ul>
<li>Now uses <code><a href="https://rdrr.io/pkg/derfinderPlot/man/vennRegions.html">derfinderPlot::vennRegions()</a></code> to show venn diagram of genomic states. Requires <code>derfinderPlot</code> version 1.3.2 or greater.</li>
<li>
<code><a href="../reference/derfinderReport.html">derfinderReport()</a></code> now has a <code>significantVar</code> argument that allows users to choose between determining significant regions by P-values, FDR adjusted P-values, or FWER adjusted P-values (if FWER adjusted P-values are absent, then FDR adjusted P-values are used instead, with a warning).</li>
</ul>
</div>
    <div id="regionreport-132" class="section level1">
<h1 class="page-header" data-toc-text="1.3.2">
<a href="#regionreport-132" class="anchor"></a>regionReport 1.3.2</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Deprecated functions with underscores in their names in favor of camelCase functions. This was done to simplify the package.</li>
</ul>
</div>
    <div id="regionreport-131" class="section level1">
<h1 class="page-header" data-toc-text="1.3.1">
<a href="#regionreport-131" class="anchor"></a>regionReport 1.3.1</h1>
<p>BUG FIXES</p>
<ul>
<li>Fixed <code><a href="../reference/renderReport.html">renderReport()</a></code> and <code><a href="../reference/derfinderReport.html">derfinderReport()</a></code> so they’ll open the correct URL when <code>interactive() == TRUE</code> and the user has knitrBootstrap version 0.9.0 installed instead of the latest GitHub version.</li>
</ul>
</div>
    <div id="regionreport-119" class="section level1">
<h1 class="page-header" data-toc-text="1.1.9">
<a href="#regionreport-119" class="anchor"></a>regionReport 1.1.9</h1>
<p>NEW FEATURES</p>
<ul>
<li>Introduced <code><a href="../reference/renderReport.html">renderReport()</a></code> which creates a simple exploratory report for any set of genomic regions. It allows the user to further customize the report by using a child file.</li>
<li>You can now use the <code>output_format</code> advanced parameter on both <code><a href="../reference/renderReport.html">renderReport()</a></code> and <code><a href="../reference/derfinderReport.html">derfinderReport()</a></code> to output a PDF file instead of an HTML file. The interactive tables are lost and only the top 20 rows are shown.</li>
</ul>
</div>
    <div id="regionreport-118" class="section level1">
<h1 class="page-header" data-toc-text="1.1.8">
<a href="#regionreport-118" class="anchor"></a>regionReport 1.1.8</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Adapted to work with <code>bumphunter</code> &gt;= 1.7.6</li>
</ul>
</div>
    <div id="regionreport-117" class="section level1">
<h1 class="page-header" data-toc-text="1.1.7">
<a href="#regionreport-117" class="anchor"></a>regionReport 1.1.7</h1>
<p>NEW FEATURES</p>
<ul>
<li>Users can now control <code>output_format</code> and <code>clean</code> options from <code><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">rmarkdown::render()</a></code> when running <code><a href="../reference/derfinderReport.html">derfinderReport()</a></code>
</li>
</ul>
</div>
    <div id="regionreport-113" class="section level1">
<h1 class="page-header" data-toc-text="1.1.3">
<a href="#regionreport-113" class="anchor"></a>regionReport 1.1.3</h1>
<p>BUG FIXES</p>
<ul>
<li>Adapted <code><a href="../reference/derfinderReport.html">derfinderReport()</a></code> to <code>derfinder</code> 1.1.5</li>
</ul>
</div>
    <div id="regionreport-0990" class="section level1">
<h1 class="page-header" data-toc-text="0.99.0">
<a href="#regionreport-0990" class="anchor"></a>regionReport 0.99.0</h1>
<p>NEW FEATURES</p>
<ul>
<li>Preparing to submit to Bioconductor.</li>
</ul>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Updated the vignette and the package to work with recent versions of the packages this package depends on.</li>
<li>Renamed the package from <code>derfinderReport</code> to <code>regionReport</code> and <code>generateReport()</code> to <code><a href="../reference/derfinderReport.html">derfinderReport()</a></code>. In the future we will add another report for a general GRanges object.</li>
<li>Simplified <code><a href="../reference/derfinderReport.html">derfinderReport()</a></code>’s call by using advanced arguments.</li>
<li>Added Travis integration.</li>
</ul>
</div>
    <div id="regionreport-0018" class="section level1">
<h1 class="page-header" data-toc-text="0.0.18">
<a href="#regionreport-0018" class="anchor"></a>regionReport 0.0.18</h1>
<ul>
<li>Now <code><a href="../reference/derfinderReport.html">derfinderReport()</a></code> has a <code>chrsStyle</code> argument to match changes in <code>derfinder</code> version 0.0.60. <code>chrsStyle</code> is set to <code>UCSC</code> by default.</li>
</ul>
</div>
    <div id="regionreport-0017" class="section level1">
<h1 class="page-header" data-toc-text="0.0.17">
<a href="#regionreport-0017" class="anchor"></a>regionReport 0.0.17</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Made more robust for cases where there is a small number of significant DERs: need at least 3 observations by chr for the chr to be included in the density plots.</li>
</ul>
</div>
    <div id="regionreport-0016" class="section level1">
<h1 class="page-header" data-toc-text="0.0.16">
<a href="#regionreport-0016" class="anchor"></a>regionReport 0.0.16</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>MA-style plots now use the scaling factor.</li>
<li>Using a GAM smoother instead of loess for MA-style plots. Helps for cases with many regions.</li>
</ul>
</div>
    <div id="regionreport-0013" class="section level1">
<h1 class="page-header" data-toc-text="0.0.13">
<a href="#regionreport-0013" class="anchor"></a>regionReport 0.0.13</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Added a vignette</li>
</ul>
</div>
    <div id="regionreport-0012" class="section level1">
<h1 class="page-header" data-toc-text="0.0.12">
<a href="#regionreport-0012" class="anchor"></a>regionReport 0.0.12</h1>
<p>BUG FIXES</p>
<ul>
<li>complying with <code>BiocCheck</code> version 1.0.0</li>
</ul>
</div>
    <div id="regionreport-0011" class="section level1">
<h1 class="page-header" data-toc-text="0.0.11">
<a href="#regionreport-0011" class="anchor"></a>regionReport 0.0.11</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>
<code>genomicState</code> data moved to <code>derfinder</code> 0.0.53</li>
</ul>
</div>
    <div id="regionreport-008" class="section level1">
<h1 class="page-header" data-toc-text="0.0.8">
<a href="#regionreport-008" class="anchor"></a>regionReport 0.0.8</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Now requires <code>knitrBootstrap</code> 1.0.0</li>
<li>Matches <code>derfinder</code> version 0.0.49</li>
</ul>
</div>
    <div id="regionreport-003" class="section level1">
<h1 class="page-header" data-toc-text="0.0.3">
<a href="#regionreport-003" class="anchor"></a>regionReport 0.0.3</h1>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul>
<li>Matches <code>derfinder</code> version 0.0.34</li>
</ul>
</div>
    <div id="regionreport-001" class="section level1">
<h1 class="page-header" data-toc-text="0.0.1">
<a href="#regionreport-001" class="anchor"></a>regionReport 0.0.1</h1>
<p>NEW FEATURES</p>
<ul>
<li>Migrated from <code>derfinder</code>
</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


