<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate a HTML/PDF report exploring DESeq2 results — DESeq2Report • regionReport</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Generate a HTML/PDF report exploring DESeq2 results — DESeq2Report"><meta property="og:description" content="This function generates a HTML report with exploratory data analysis plots
for DESeq2 results created with DESeq. Other output formats
are possible such as PDF but lose the interactivity. Users can easily append
to the report by providing a R Markdown file to customCode, or can
customize the entire template by providing an R Markdown file to
template."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">regionReport</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.35.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/regionReport.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/bumphunterExample.html">Example report using bumphunter results</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/leekgroup/regionReport/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate a HTML/PDF report exploring DESeq2 results</h1>
    <small class="dont-index">Source: <a href="https://github.com/leekgroup/regionReport/blob/HEAD/R/DESeq2Report.R" class="external-link"><code>R/DESeq2Report.R</code></a></small>
    <div class="hidden name"><code>DESeq2Report.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function generates a HTML report with exploratory data analysis plots
for DESeq2 results created with <a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a>. Other output formats
are possible such as PDF but lose the interactivity. Users can easily append
to the report by providing a R Markdown file to <code>customCode</code>, or can
customize the entire template by providing an R Markdown file to
<code>template</code>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">DESeq2Report</span><span class="op">(</span></span>
<span>  <span class="va">dds</span>,</span>
<span>  project <span class="op">=</span> <span class="st">""</span>,</span>
<span>  <span class="va">intgroup</span>,</span>
<span>  colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  res <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  nBest <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  nBestFeatures <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  customCode <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  outdir <span class="op">=</span> <span class="st">"DESeq2Exploration"</span>,</span>
<span>  output <span class="op">=</span> <span class="st">"DESeq2Exploration"</span>,</span>
<span>  browse <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  device <span class="op">=</span> <span class="st">"png"</span>,</span>
<span>  template <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  searchURL <span class="op">=</span> <span class="st">"http://www.ncbi.nlm.nih.gov/gene/?term="</span>,</span>
<span>  theme <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  digits <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dds</dt>
<dd><p>A <a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a> object with the results from
running <a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a>.</p></dd>


<dt>project</dt>
<dd><p>The title of the project.</p></dd>


<dt>intgroup</dt>
<dd><p>interesting groups: a character vector of names in
<code>colData(x)</code> to use for grouping. This parameter is passed to functions
such as <a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a>.</p></dd>


<dt>colors</dt>
<dd><p>vector of colors used in heatmap. If <code>NULL</code>, then a
a default set of colors will be used. This argument is passed to
<a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a>.</p></dd>


<dt>res</dt>
<dd><p>A <a href="https://rdrr.io/pkg/DESeq2/man/DESeqResults.html" class="external-link">DESeqResults</a> object. If <code>NULL</code>, then
<a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a> will be used on <code>dds</code> with default parameters.</p></dd>


<dt>nBest</dt>
<dd><p>The number of features to include in the interactive
table. Features are ordered by their adjusted p-values.</p></dd>


<dt>nBestFeatures</dt>
<dd><p>The number of best features to make plots of their
counts. We recommend a small number, say 20.</p></dd>


<dt>customCode</dt>
<dd><p>An absolute path to a child R Markdown file with code to be
evaluated before the reproducibility section. Its useful for users who want
to customize the report by adding conclusions derived from the data and/or
further quality checks and plots.</p></dd>


<dt>outdir</dt>
<dd><p>The name of output directory.</p></dd>


<dt>output</dt>
<dd><p>The name of output HTML file (without the html extension).</p></dd>


<dt>browse</dt>
<dd><p>If <code>TRUE</code> the HTML report is opened in your browser once
it's completed.</p></dd>


<dt>device</dt>
<dd><p>The graphical device used when knitting. See more at
<a href="http://yihui.name/knitr/options" class="external-link">http://yihui.name/knitr/options</a> (<code>dev</code> argument).</p></dd>


<dt>template</dt>
<dd><p>Template file to use for the report. If not provided, will
use the default file found in DESeq2Exploration/DESeq2Exploration.Rmd
within the package source.</p></dd>


<dt>searchURL</dt>
<dd><p>A url used for searching the name of the features in
the web. By default <code>http://www.ncbi.nlm.nih.gov/gene/?term=</code> is used
which is the recommended option when features are genes. It's only used
when the output is a HTML file.</p></dd>


<dt>theme</dt>
<dd><p>A ggplot2 <a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a> to use for the plots made with
ggplot2.</p></dd>


<dt>digits</dt>
<dd><p>The number of digits to round to in the interactive table of
the top <code>nBestFeatures</code>. Note that p-values and adjusted p-values won't
be rounded.</p></dd>


<dt>...</dt>
<dd><p>Arguments passed to other methods and/or advanced arguments.
Advanced arguments:</p><dl><dt>software </dt>
<dd><p>The name of the package used for performing the
differential expression analysis. Either <code>DESeq2</code> or <code>edgeR</code>.</p></dd>

<dt>dge </dt>
<dd><p>A <a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a> object. <code>NULL</code> by default and only
used by <a href="edgeReport.html">edgeReport</a>.</p></dd>

<dt>theCall </dt>
<dd><p>The function call. <code>NULL</code> by default and only used by
<a href="edgeReport.html">edgeReport</a>.</p></dd>

<dt>output_format </dt>
<dd><p>Either <code>html_document</code>, <code>pdf_document</code> or
<code><a href="https://rdrr.io/pkg/knitrBootstrap/man/bootstrap_document.html" class="external-link">knitrBootstrap::bootstrap_document</a></code> unless you modify the YAML
template.</p></dd>

<dt>clean </dt>
<dd><p>Logical, whether to clean the results or not. Passed to
<a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a>.</p></dd>


</dl></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>An HTML report with a basic exploration for the given set of DESeq2
results. See an example at <a href="http://leekgroup.github.io/regionReport/reference/DESeq2Report-example/DESeq2Exploration.html" class="external-link">http://leekgroup.github.io/regionReport/reference/DESeq2Report-example/DESeq2Exploration.html</a>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Set <code>output_format</code> to <code>'knitrBootstrap::bootstrap_document'</code> or
<code>'pdf_document'</code> if you want a HTML report styled by knitrBootstrap or
a PDF report respectively. If using knitrBootstrap, we recommend the version
available only via GitHub at <a href="https://github.com/jimhester/knitrBootstrap" class="external-link">https://github.com/jimhester/knitrBootstrap</a>
which has nicer features than the current version available via CRAN. You can
also set the <code>output_format</code> to <code>'html_document'</code> for a HTML
report styled by rmarkdown. The default is set to
<code>'BiocStyle::html_document'</code>.</p>
<p>If you modify the YAML front matter of <code>template</code>, you can use other
values for <code>output_format</code>.</p>
<p>The HTML report styled with knitrBootstrap can be smaller in size than the
<code>'html_document'</code> report.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Leonardo Collado-Torres</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Load example data from the pasilla package as done in the DESeq2 vignette</span></span></span>
<span class="r-in"><span><span class="co">## at</span></span></span>
<span class="r-in"><span><span class="co">## &lt;https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#count-matrix-input&gt;.</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"pasilla"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: DEXSeq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocParallel</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biobase</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘BiocGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:stats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     tapply, union, unique, unsplit, which.max, which.min</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Welcome to Bioconductor</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Vignettes contain introductory material; view with</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: SummarizedExperiment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: MatrixGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: matrixStats</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘matrixStats’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:Biobase’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     anyMissing, rowMedians</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘MatrixGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:matrixStats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedSds, rowWeightedVars</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:Biobase’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMedians</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: stats4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: S4Vectors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘S4Vectors’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:utils’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     findMatches</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     I, expand.grid, unname</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: IRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: DESeq2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: AnnotationDbi</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: RColorBrewer</span>
<span class="r-in"><span><span class="va">pasCts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span></span>
<span class="r-in"><span>    <span class="st">"pasilla_gene_counts.tsv"</span>,</span></span>
<span class="r-in"><span>    package <span class="op">=</span> <span class="st">"pasilla"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pasAnno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span></span>
<span class="r-in"><span>    <span class="st">"pasilla_sample_annotation.csv"</span>,</span></span>
<span class="r-in"><span>    package <span class="op">=</span> <span class="st">"pasilla"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">cts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">pasCts</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, row.names <span class="op">=</span> <span class="st">"gene_id"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">coldata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">pasAnno</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">coldata</span> <span class="op">&lt;-</span> <span class="va">coldata</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"type"</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">coldata</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">$</span><span class="va">condition</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">coldata</span><span class="op">$</span><span class="va">type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">$</span><span class="va">type</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"fb"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">cts</span> <span class="op">&lt;-</span> <span class="va">cts</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create DESeqDataSet object from the pasilla package</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/mikelove/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSetFromMatrix</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    countData <span class="op">=</span> <span class="va">cts</span>,</span></span>
<span class="r-in"><span>    colData <span class="op">=</span> <span class="va">coldata</span>,</span></span>
<span class="r-in"><span>    design <span class="op">=</span> <span class="op">~</span><span class="va">condition</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating size factors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The output will be saved in the 'DESeq2Report-example' directory</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="st">"DESeq2Report-example"</span>, showWarnings <span class="op">=</span> <span class="cn">FALSE</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Generate the HTML report</span></span></span>
<span class="r-in"><span><span class="va">report</span> <span class="op">&lt;-</span> <span class="fu">DESeq2Report</span><span class="op">(</span><span class="va">dds</span>, <span class="st">"DESeq2-example"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"type"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    outdir <span class="op">=</span> <span class="st">"DESeq2Report-example"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Writing 9 Bibtex entries ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Results written to file ‘DESeq2Report-example/DESeq2Exploration.bib’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> processing file: DESeq2Exploration.Rmd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                          |                                                  |   0%  |                                                          |.                                                 |   2%                      |                                                          |..                                                |   4% (docSetup)           |                                                          |...                                               |   6%                      |                                                          |....                                              |   9% (setup)              |                                                          |.....                                             |  11%                      |                                                          |......                                            |  13% (PCA)                |                                                          |.......                                           |  15%                      |                                                          |.........                                         |  17% (sampleDist)         |                                                          |..........                                        |  19%                      |                                                          |...........                                       |  21% (MAplotalpha)        |                                                          |............                                      |  23%                      |                                                          |.............                                     |  26% (MAplotalphaHalf)    |                                                          |..............                                    |  28%                      |                                                          |...............                                   |  30% (MAplotalpha-nBest)  |                                                          |................                                  |  32%                      |                                                          |.................                                 |  34% (pvalueHistogram)    |                                                          |..................                                |  36%                      |                                                          |...................                               |  38% (pvalueSumm)         |                                                          |....................                              |  40%                      |                                                          |.....................                             |  43% (pvalueTable)        |                                                          |......................                            |  45%                      |                                                          |.......................                           |  47% (padjHistogram)      |                                                          |........................                          |  49%                      |                                                          |..........................                        |  51% (padjSumm)           |                                                          |...........................                       |  53%                      |                                                          |............................                      |  55% (padjTable)          |                                                          |.............................                     |  57%                      |                                                          |..............................                    |  60% (topFeatures)        |                                                          |...............................                   |  62%                      |                                                          |................................                  |  64% (plotCounts)         |                                                          |.................................                 |  66%                      |                                                          |..................................                |  68% (edgeR-BCV)          |                                                          |...................................               |  70%                      |                                                          |....................................              |  72% (edgeR-MDS)          |                                                          |.....................................             |  74%                      |                                                          |......................................            |  77% (unnamed-chunk-1)    |                                                          |.......................................           |  79%                      |                                                          |........................................          |  81% (thecall)            |                                                          |.........................................         |  83%                      |                                                          |...........................................       |  85% (reproducibility1)   |                                                          |............................................      |  87%                      |                                                          |.............................................     |  89% (reproducibility2)   |                                                          |..............................................    |  91%                      |                                                          |...............................................   |  94% (reproducibility3)   |                                                          |................................................  |  96%                      |                                                          |................................................. |  98% (bibliography)       |                                                          |..................................................| 100%                                                                                                                                  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> output file: DESeq2Exploration.knit.md</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> /usr/local/bin/pandoc +RTS -K512m -RTS DESeq2Exploration.knit.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output DESeq2Exploration.html --lua-filter /__w/_temp/Library/bookdown/rmarkdown/lua/custom-environment.lua --lua-filter /usr/local/lib/R/site-library/rmarkdown/rmarkdown/lua/pagebreak.lua --lua-filter /usr/local/lib/R/site-library/rmarkdown/rmarkdown/lua/latex-div.lua --embed-resources --standalone --wrap preserve --variable bs3=TRUE --section-divs --table-of-contents --toc-depth 3 --variable toc_float=1 --variable toc_selectors=h1,h2,h3 --variable toc_collapsed=1 --variable toc_smooth_scroll=1 --variable toc_print=1 --template /tmp/RtmpdZclSV/BiocStyle/template.html --no-highlight --variable highlightjs=1 --number-sections --variable theme=bootstrap --css /__w/_temp/Library/BiocStyle/resources/html/bioconductor.css --mathjax --variable 'mathjax-url=https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --include-in-header /tmp/RtmpdZclSV/rmarkdown-str29f235d159bd.html --variable code_folding=hide --variable code_menu=1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Output created: DESeq2Exploration.html</span>
<span class="r-plt img"><img src="DESeq2Report-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="co">## Browse the report</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL</a></span><span class="op">(</span><span class="va">report</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## See the example output at</span></span></span>
<span class="r-in"><span><span class="co">## http://leekgroup.github.io/regionReport/reference/DESeq2Report-example/DESeq2Exploration.html</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="co">## Note that you can run the example using:</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/example.html" class="external-link">example</a></span><span class="op">(</span><span class="st">"DESeq2Report"</span>, <span class="st">"regionReport"</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

